### PR TITLE
general: minor changes to make import to Google easier

### PR DESCRIPTION
**Pull request by jmarantz**
*Description*: Changes a few coding cases to make import cleaner.
*Risk Level*: low
*Testing*: //test/...
*Docs Changes*: n/a
*Release Notes*: n/a

### CODE REVIEW

File: test/test_common/utility.cc [hunk 1]
```diff
@@ -227,9 +227,9 @@ void TestUtility::createSymlink(const std::string& target, const std::string& li
 // static
 absl::Time TestUtility::parseTime(const std::string& input, const std::string& input_format) {
   absl::Time time;
-  std::string error;
-  EXPECT_TRUE(absl::ParseTime(input_format, input, &time, &error))
-      << " error \"" << error << "\" from failing to parse timestamp \"" << input
+  std::string absl_string_error; // Name triggers Google import process to sed to a 'string'.
```
[the remark this answers is not in the document]
Comment (jmarantz) [reply test/test_common/utility.cc]: I want to put into our sed-script something like
```
sed -e -s /std::string::absl_string_/string absl::string_/g
```

But now I have a better idea...I'll put this into a 'using' type nickname and sed just that file.  Hang on.

File: test/common/thread/thread_factory_singleton_test.cc [hunk 2]
```diff
@@ -9,26 +9,25 @@ namespace Envoy {
...
   // code under test.
-  std::thread thread([&singleton_test]() {
-    EXPECT_DEBUG_DEATH(singleton_test.checkThreadId(),
-                       "assert failure: run_tid_->isCurrentThreadId()");
-  });
+  bool is_current = checkThreadId();
+  EXPECT_TRUE(is_current);
+  std::thread thread([this, &is_current]() { is_current = checkThreadId(); });
   thread.join();
+  EXPECT_FALSE(is_current) << "assert failure: run_tid_->isCurrentThreadId()";
```
Comment (AndresGuedez) [inline test/common/thread/thread_factory_singleton_test.cc]: nit: the message is misleading since this is no longer testing an ASSERT().

**Comment by ggreenway** [comment]
This change makes me a bit uncomfortable. 

On the one hand, Google is a heavy contributor to this project and has added huge amounts of value to this project, it's features, and it's robustness.

On the other hand, Google-specific implementation details are leaking into the OSS project. For non-Googlers, it is not at all clear which places should use the GoogleStringHack vs just doing normal things.

There's always a balance to be struck between pragmatism and purity/perfection. This change feels like it might be tipping too far. But it's not so far that I would put my foot down and say this cannot be done. But I think it's worth discussing at least.

**Comment by jmarantz** [comment]
A fair point, but there are already a few different places where similar hacks are done in the Envoy codebase. I was offering this as a slightly more obvious form of that.

Another option is to do essentially the same thing, but instead of calling it GoogleStringHack, call it AbslString. Would that feel better?

**Comment by mattklein123** [comment]
I'm not a fan of this change either. The "google string issue" has been a source of friction for quite some time. My understanding was that Google is fixing this internally "eventually." Until that time, can't we just keep doing what we have been doing with Google coming back and issuing fix PRs when necessary? I don't see anyone actually using the `GoogleStringHack` type when they write code anyway, so it seems like fix PRs will be needed no matter what.

File: test/test_common/utility.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmarantz) [reply test/test_common/utility.cc]: OK, parse_error.

**Comment by ggreenway** [comment]
Thanks!

**Pull request merged**